docs/txp.md

```diff
@@ -0,0 +1,434 @@
+   because data is unknown. This behaviour depends on which type of
+   data we are processing, each particular case is described in more
+   details below.
+
+The behaviour depends on two protocol versions: version used by this
+software and last adopted version. We verify that data in blocks is
+known if protocol version used by this software is greater than or
+equal to the adopted version. That's because in this case:
+
+1. Authors of this software are aware of the adopted version.
+2. Each issued block must be formed with respect to adopted version.
+
+Comparison is quite tricky here. Table below demonstrates it.
+
+| Our   | Adopted | Check? |
+| ----- | ------- | ------ |
+| 1.2.3 |  1.2.3  | Yes    |
+| 1.2.3 |  1.2.4  | No     |
+| 1.2.3 |  1.2.2  | No     |
+| 1.2.3 |  1.3.2  | No     |
+| 1.2.3 |  1.1.1  | Yes    |
+| 2.2.8 |  1.9.9  | Yes    |
+
+If `(major, minor)` of our version is greater than of adopted
+one, then check is certainly done. If it's equal, then check is
+done only if `alt` component is the same as adopted one. In
+other cases (i. e. when our `(major, minor)` is less than from
+adopted version) check is not done.
+
+Note: when we say that attributes must be known, it means that
+unparsed fields must be empty.
+
+### Verification
+
+As stated above, we want to verify block `B` validity against GState `S`.
+To do so, each transaction from block's transaction payload is taken separately (in order of inclusion into block) and considered against `S`.
+
+Let's assume that version comparison as described at the end of previos section (Unknown data handling) is done.
+Let's denote flag `verifyAllIsKnown` and assign it value:
+ * `True` if check is to be done
+    - I.e. only known data should be considered valid (unknown data is prohibitied)
+ * `False` otherwise
+
+#### General checks
+
+Checks, that are considering whole transaction, not inputs/outputs in particular.
+
+- *Consistency check*: number of inputs in a transaction must be the same as number of witnesses for this transaction
+- *Attributes known check*: if `verifyAllIsKnown` is `True`, all transaction attributes must be known.
+- *Transaction size check*: size of transaction doesn't exceed size limit
+    - Transaction size is computed as number of bytes in serialized `TxAux` (which contains transaction and its witness)
+    - Transaction size limit is part of `BlockVersionData`.
+    - If the size of transaction is greater than the limit from the adopted `BlockVersionData`, transaction is
+invalid.
+- *Bootstrap era check*: if the block was created during bootstrap era, all
+output addresses must have `BootstrapEraDistr` distribution
+  - Predicate "was created during bootstrap era" is checked with block's slot field
+    and adopted `BlockVersionData` (which contains data which slot is defined last for bootstrap era)
+
+#### Outputs checks
+
+For each output address we perform the following checks:
```

Comment:
@gromakovsky is this outputs check list complete?

I don't see an *important* check that all outputs are different.

CC @neongreen